### ISSUE TITLE
Suggested license cleanup

### ISSUE THREAD

**Issue by GitMensch** [mridoni/gix#2]
* explicit [state the license in the README](https://www.gnu.org/licenses/gpl-faq.html#LicenseCopyOnly)
* add [a license notice to each source](https://www.gnu.org/licenses/gpl-faq.html#NoticeInSourceFile)
* possibly in both places: distribute under "GPLv2 or newer"

**Comment by mridoni** [comment]
I have added the license header for GPL2+ to all the files, now I'm manually reviewing them one by one to check for exceptions and errors before committing. Files in QtWebApp do not have a license header at the top, just a LICENSE file added to the main repository. Would it be ok to add a header to each of these files (obviously with proper attribution to the copyright owner, Stefan Frings) to clarify attribution?

**Comment by mridoni** [comment]
I have added license headers to all the files, with proper attribution. In gix-http, QtWebApp files that are unchanged (most of them) only have Stefan Fring's copyright. Files that have been heavily modified or are derived from QtWebApp have both copyright notices (Stefan Fring's and mine). The same applies to GixSQL libraries and preprocessor.

All license files have been added to the repository.

Since I am far from an expert when it comes to licensing, I am leaving this and #11 open for a few days, so comment and suggestions can be added.

**Comment by GitMensch** [comment]
Thank you for taking this serious as shown with 6fec972c6a19dbb3226a5bc48ca6d054fffa6985! These changes already solve most of the issues I've seen.
I also wouldn't claim to be a license expert and I'm no lawyer but here are my thoughts (it looks much but "we're nearly there")...

License issues that need to be adjusted:
* as QWebApp was "forked" from [version 1.7.8](https://github.com/mridoni/gix/blob/main/gix-http/http/httpglobal.cpp) the license notes for it should say `2010-2019` instead of `2010`
* as I understand, everything that uses a current version of Qt needs either to have a commercial license of Qt or be (L)GPLv3 / (L)GPLv3+ licensed, see https://www.qt.io/faq/tag/qt-open-source-licensing, if that is correct then Gix-IDE must be under GPLv3(+) and its own libraries that use QT need to be LGPLv3(+)
* *if* gix-sql also uses Qt-components it is likely reasonable to move it from LGPLv2+ to LGPLv3+ (or remove the QT parts if those aren't used a lot; could be reasonable if you intend to have it working also without a current LGPLv3+'ed libcob), if it doesn't then it is fine to leave it as-is.
* from the licensing entry in the README it seems that Gix-IDE is "GPLv2 only" (which seems not the case according to the file headers with say GPLv2+)
* even without a recent QT (haven't checked, but its version 3 or 4 should be LGPLv2+) I think Gix-IDE must be licensed under GPLv3+ (or GPLv3only) as it links to QtWebApp and its [version 3 is not compatible to version 2](https://www.gnu.org/licenses/gpl-faq.html#v2v3Compatibility)

README:
* as the referenced "fork" of ocesql has only one single change (changed 4 characters alltogether) I recommend to instead describe it as fork of https://github.com/opensourcecobol/Open-COBOL-ESQL (Version 1.0)
* I suggest to keep the attribution out of the license part, instead have those in the ESQL (where it is in partially) and in the HTTP part
* minor typo: the title of the README has "iDE" -> "IDE"

Technical: Instead of an additional commit you may want to amend your previous change (= have one commit with the license adjustments instead of two) and force-push the repo.
... **if** you change the main license (for Gix-IDE) I suggest to delete the License file in the repo with the same commit and then recreate it via GitHub in the correct version, this should also set the metadata for the repo correct (the license that is shown on the repo main page which currently only says "View License").

**Comment by mridoni** [comment]
Thanks for the feedback, this is really appreciated. I'll proceed in the next few days, when my day job allows for it. 

Two things: 

- I was also worried about possible incompatibilities between GPL v2 and v3, but the sources I found (mainly forum posts) pointed me in another (read: wrong) direction. The source you provided, and that I couldn't find is obviously authoritative. Due to this and to the Qt licensing issue, I will "upgrade" the license to (L)GPL v3 for Gix-IDE and GixSQL (the original OceSQL license allows for distribution under a "later" version so that should be ok anyway.
- I just referenced that fork of ocesql because I couldn't locate the original address (yes, I should refine my Internet search skills :-) ) so I will definitely change the link to the correct one.

Thanks
